lib/helpers/pluralize.js

```diff
@@ -6,31 +6,36 @@ module.exports = pluralize;
  * Pluralization rules.
  */
 
+// David Xie: rebase pluralization rules from pluralize.js committed on Oct 6, 2021
+// License of pluralize.js: MIT License
+// Commit: https://github.com/plurals/pluralize/blob/36f03cd2d573fa6d23e12e1529fa4627e2af74b4/pluralize.js
+
 exports.pluralization = [
-  [/(m)an$/gi, '$1en'],
-  [/(pe)rson$/gi, '$1ople'],
-  [/(child)$/gi, '$1ren'],
-  [/^(ox)$/gi, '$1en'],
-  [/(ax|test)is$/gi, '$1es'],
-  [/(octop|vir)us$/gi, '$1i'],
-  [/(alias|status)$/gi, '$1es'],
-  [/(bu)s$/gi, '$1ses'],
-  [/(buffal|tomat|potat)o$/gi, '$1oes'],
-  [/([ti])um$/gi, '$1a'],
-  [/sis$/gi, 'ses'],
-  [/(?:([^f])fe|([lr])f)$/gi, '$1$2ves'],
-  [/(hive)$/gi, '$1s'],
-  [/([^aeiouy]|qu)y$/gi, '$1ies'],
-  [/(x|ch|ss|sh)$/gi, '$1es'],
-  [/(matr|vert|ind)ix|ex$/gi, '$1ices'],
-  [/([m|l])ouse$/gi, '$1ice'],
-  [/(kn|w|l)ife$/gi, '$1ives'],
-  [/(quiz)$/gi, '$1zes'],
-  [/^goose$/i, 'geese'],
-  [/s$/gi, 's'],
-  [/([^a-z])$/, '$1'],
-  [/$/gi, 's']
-];
+  [/s?$/i, 's'],
```

Comment:
We unfortunately cannot merge this change until Mongoose 7, because of all of the backwards breaking changes here.